### PR TITLE
Endpoint structure

### PR DESCRIPTION
**Pull request by joshganote**
- Implementing a common server structure with Models and Messages. 
- Creating the endpoint structure to return mock data from that endpoint.
- Integrate the profile view with the newly created profile view api.
- Replace the stub service calls with live service calls

### CODE REVIEW

**Comment by rogisolorzano** [comment]
I wasn't able to get the project running, I got this error that `$mat-lightBlue` is undefined.
Screen Shot 2021-01-30 at 5 46 24 AM
`$mat-lightBlue` is not a part of the angular theming library being imported, I changed this to `$mat-light-blue` locally to fix it. Did you perhaps make some updates inside this `~@angular/material/theming` file?
Screen Shot 2021-01-30 at 5 50 01 AM
This file is a part of the `@angular/material` package so all changes would get lost whenever you run `npm install` again.

After changing the scss variable to get the project running, I'm getting this error and the page is black.
Screen Shot 2021-01-30 at 5 44 39 AM
Am I maybe missing some steps to get this running that could be added to the README?

File: server/Model/IUserDto.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export class User {
```
Comment (Devereux-Henley) [inline server/Model/IUserDto.ts:1]: Is this being used?

File: server/Data/UserController.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export class UserController {
+  constructor(private readonly formService: UserService) {}
+
+  @Get(':id')
+  getUser(@Param('id') id: string): IUser {
+    const user = this.formService.getUser(id);
+    return { ...user };
+  }
+
+  @Patch(':id')
```
Comment (Devereux-Henley) [inline server/Data/UserController.ts:17]: Would use `PUT` here. PUT is used for full resource replacement, while `PATCH` is used for passing instructions on how to update a resource (partially or with a specified operation).

File: server/Data/UserController.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Controller, Get, Body, Param, Patch } from '@nestjs/common';
+import { UserService } from './IUserDao';
+import { IUser } from '../Model/IUser';
+import { User } from '../Model/IUserDto';
+
+// get rid of userform names. just user in the backend
```
Comment (Devereux-Henley) [inline server/Data/UserController.ts:6]: Would remove TODO comments.

File: server/App/app.service.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { Injectable } from '@nestjs/common';
```
Comment (Devereux-Henley) [inline server/App/app.service.ts:1]: Can remove the `App` folder.

File: server/Data/IUserDao.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { IUser } from '../Model/IUser';
+import { MockUser } from '../Model/MockUser';
+
+@Injectable()
+export class UserService {
+  userData = MockUser;
+
+  private readonly userArray: IUser[] = this.userData;
```
Comment (Devereux-Henley) [inline server/Data/IUserDao.ts:9]: Local initialization should be done in a constructor.

File: server/Data/IUserDao.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { IUser } from '../Model/IUser';
+import { MockUser } from '../Model/MockUser';
+
+@Injectable()
+export class UserService {
+  userData = MockUser;
```
Comment (Devereux-Henley) [inline server/Data/IUserDao.ts:7]: Do we need this var?

File: server/Model/IUserDto.ts (hunk 1 above)
Comment (joshganote) [reply server/Model/IUserDto.ts:1]: It's being used to @Body decorator to define the expected data object. Probably explaining that wrong but I guess that could also be an example of what a Message should be.

server/data/UserController
`@Patch(':id')
  updateUser(@Body() createUserDto: User, @Param('id') id: string): IUser {
    const updatedUser = this.formService.updateUser(id, createUserDto);
    return { ...updatedUser };
  }`

File: src/app/services/user-data.service.ts [hunk 7]
```diff
@@ -3,22 +3,29 @@ import { IFormData } from './data/IFormData';
...
-  constructor() {
+  constructor(private http: HttpClient) {
     this.data = { ...FormData };
   }
 
-  getData(): Observable<IFormData> {
-    return of(FormData).pipe(delay(1000));
+  getData(id: string): Observable<IFormData> {
+    of(FormData).pipe(delay(1000));
+    return this.http.get<IFormData>(`http://localhost:4200/api/user/${id}`);
```
Comment (Devereux-Henley) [inline src/app/services/user-data.service.ts:19]: Uris should be configured - src/environments/environment.${environment}.ts

Would put the localhost uri in environment.ts for now as that is the default - can update types of builds later.

File: server/main.ts [hunk 8]
```diff
@@ -3,16 +3,20 @@ import { AppModule } from './app.module';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
+  app.enableCors({
+    origin: 'http://localhost:4200',
```
Comment (Devereux-Henley) [inline server/main.ts:7]: uri's should be configured. Would follow one of the patterns here for setting this configuration: https://docs.nestjs.com/techniques/configuration

**Review by Devereux-Henley — APPROVED** [review]

**Pull request merged**